IPython/terminal/ptshell.py

```diff
@@ -232,7 +230,10 @@ def _(event):
         self._style = self._make_style_from_name(self.highlighting_style)
         style = DynamicStyle(lambda: self._style)
 
+        editing_mode = EditingMode.VI if self.vi_mode else EditingMode.EMACS
```

Comment:
Are there more modes on the cards? We haven't released this yet, so we could still change from a boolean to a string to allow future extensions.